-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-34219/23558

ERROR: Build errors found during clang-tidy run.
```
Geometry/MuonNumbering/plugins/MuonGeometryConstantsESModule.cc:35:1: error: version control conflict marker in file [clang-diagnostic-error]
<<<<<<< HEAD
^
Suppressed 2481 warnings (2480 in non-user code, 1 with check filters).
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:129: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/Configuration/Applications/python/ConfigBuilder.py
+++ b/Configuration/Applications/python/ConfigBuilder.py
@@ -763,7 +763,9 @@ class ConfigBuilder(object):
                 if ('SIM' in self.stepMap or 'reSIM' in self.stepMap) and not self._options.fast:
                     self.loadAndRemember(self.SimGeometryCFF)
                     if self.geometryDBLabel:
-                        self.executeAndRemember('process.XMLFromDBSource.label = cms.string("%s")'%(self.geometryDBLabel))
+                        self.executeAndRemember('if hasattr(process, "XMLFromDBSource"): process.XMLFromDBSource.label="%s"'%(self.geometryDBLabel))
+                        self.executeAndRemember('if hasattr(process, "DDDetectorESProducerFromDB"): process.DDDetectorESProducerFromDB.label="%s"'%(self.geometryDBLabel))
+
         except ImportError:
             print("Geometry option",self._options.geometry,"unknown.")
             raise

--- a/Configuration/Geometry/python/GeometryDD4hepSimDB_cff.py
+++ b/Configuration/Geometry/python/GeometryDD4hepSimDB_cff.py
@@ -1,0 +1,26 @@
+import FWCore.ParameterSet.Config as cms
+
+#
+# Geometry master configuration
+#
+# Ideal geometry, needed for simulation
+
+DDSpecParRegistryESProducer = cms.ESProducer("DDSpecParRegistryESProducer",
+                                             appendToDataLabel = cms.string('')
+)
+
+DDVectorRegistryESProducer = cms.ESProducer("DDVectorRegistryESProducer",
+                                            appendToDataLabel = cms.string(''))
+
+DDCompactViewESProducer = cms.ESProducer("DDCompactViewESProducer",
+                                         appendToDataLabel = cms.string('')
+)
+
+from DetectorDescription.DDCMS.DDDetectorESProducerFromDB_cfi import *
+
+from Geometry.TrackerNumberingBuilder.trackerNumberingGeometryDB_cfi import *
+
+from Geometry.EcalCommonData.ecalSimulationParameters_cff   import *
+from Geometry.HcalCommonData.hcalSimDBConstants_cff         import *
+
+from Geometry.MuonNumbering.muonGeometryConstants_cff       import *

--- a/Configuration/Geometry/python/GeometryDD4hepSimDB_cff.py
+++ b/Configuration/Geometry/python/GeometryDD4hepSimDB_cff.py
@@ -24,3 +24,4 @@ from Geometry.EcalCommonData.ecalSimulationParameters_cff   import *
 from Geometry.HcalCommonData.hcalSimDBConstants_cff         import *
 
 from Geometry.MuonNumbering.muonGeometryConstants_cff       import *
+from Geometry.MuonNumbering.muonOffsetESProducer_cff        import *

--- a/Configuration/Geometry/python/GeometryDDDSimDB_cff.py
+++ b/Configuration/Geometry/python/GeometryDDDSimDB_cff.py
@@ -1,0 +1,15 @@
+import FWCore.ParameterSet.Config as cms
+
+#
+# Geometry master configuration
+#
+# Ideal geometry, needed for simulation
+from GeometryReaders.XMLIdealGeometryESSource.cmsGeometryDB_cff import *
+from Geometry.MuonNumbering.muonNumberingInitialization_cfi import *
+from Geometry.TrackerNumberingBuilder.trackerNumberingGeometryDB_cfi import *
+from Geometry.EcalCommonData.ecalSimulationParameters_cff   import *
+from Geometry.HcalCommonData.hcalSimulationParameters_cfi   import *
+from Geometry.HcalCommonData.hcalDDDSimConstants_cfi        import *
+from Geometry.HcalCommonData.hcalSimulationConstants_cfi    import *
+from Geometry.HcalCommonData.caloSimulationParameters_cff   import *
+from Geometry.MuonNumbering.muonGeometryConstants_cff       import *

--- a/Configuration/Geometry/python/GeometryRecoDB_cff.py
+++ b/Configuration/Geometry/python/GeometryRecoDB_cff.py
@@ -17,8 +17,7 @@ from Geometry.CaloEventSetup.AlignedCaloGeometryDBReader_cfi import *
 from Geometry.CaloEventSetup.EcalTrigTowerConstituents_cfi import *
 from Geometry.EcalMapping.EcalMapping_cfi import *
 from Geometry.EcalMapping.EcalMappingRecord_cfi import *
-from Geometry.HcalCommonData.hcalDDDSimConstants_cfi import *
-from Geometry.HcalCommonData.hcalDDDRecConstants_cfi import *
+from Geometry.HcalCommonData.hcalDBConstants_cff import *
 from Geometry.HcalEventSetup.hcalTopologyIdeal_cfi import *
 
 #  Alignment

--- a/Configuration/Geometry/python/GeometrySimDB_cff.py
+++ b/Configuration/Geometry/python/GeometrySimDB_cff.py
@@ -4,12 +4,14 @@ import FWCore.ParameterSet.Config as cms
 # Geometry master configuration
 #
 # Ideal geometry, needed for simulation
-from GeometryReaders.XMLIdealGeometryESSource.cmsGeometryDB_cff import *
-from Geometry.MuonNumbering.muonNumberingInitialization_cfi import *
-from Geometry.TrackerNumberingBuilder.trackerNumberingGeometryDB_cfi import *
-from Geometry.EcalCommonData.ecalSimulationParameters_cff   import *
-from Geometry.HcalCommonData.hcalSimulationParameters_cfi   import *
-from Geometry.HcalCommonData.hcalDDDSimConstants_cfi        import *
-from Geometry.HcalCommonData.hcalSimulationConstants_cfi    import *
-from Geometry.HcalCommonData.caloSimulationParameters_cff   import *
-from Geometry.MuonNumbering.muonGeometryConstants_cff       import *
+
+from Configuration.ProcessModifiers.dd4hep_cff import dd4hep
+
+def _loadSimGeomDBDDD(process) :
+    process.load('Configuration.Geometry.GeometryDDDSimDB_cff')
+
+def _loadSimGeomDBDD4hep(process) :
+    process.load('Configuration.Geometry.GeometryDD4hepSimDB_cff')
+
+modifyGeometryConfiguration = (~dd4hep).makeProcessModifier(_loadSimGeomDBDDD)
+modifyGeometryConfiguration_dd4hep = dd4hep.makeProcessModifier(_loadSimGeomDBDD4hep)

--- a/DetectorDescription/DDCMS/plugins/DDDetectorESProducer.cc
+++ b/DetectorDescription/DDCMS/plugins/DDDetectorESProducer.cc
@@ -79,8 +79,7 @@ DDDetectorESProducer::DDDetectorESProducer(const ParameterSet& iConfig)
     }
     findingRecord<IdealMagneticFieldRecord>();
   } else {
-    auto c = setWhatProduced(
-        this, &DDDetectorESProducer::produceGeom, edm::es::Label(iConfig.getParameter<std::string>("@module_label")));
+    auto c = setWhatProduced(this, &DDDetectorESProducer::produceGeom);
     if (fromDB_) {
       geomToken_ = c.consumes(edm::ESInputTag("", label_));
     }

--- a/DetectorDescription/DDCMS/src/ES_DDDetector.cc
+++ b/DetectorDescription/DDCMS/src/ES_DDDetector.cc
@@ -8,7 +8,7 @@
 
 using namespace cms;
 
-TYPELOOKUP_DATA_REG(DDCompactView);
+TYPELOOKUP_DATA_REG(cms::DDCompactView);
 TYPELOOKUP_DATA_REG(DDDetector);
 TYPELOOKUP_DATA_REG(dd4hep::SpecParRegistry);
 TYPELOOKUP_DATA_REG(DDVectorRegistry);

--- a/Geometry/HcalCommonData/python/hcalSimDBConstants_cff.py
+++ b/Geometry/HcalCommonData/python/hcalSimDBConstants_cff.py
@@ -1,0 +1,8 @@
+import FWCore.ParameterSet.Config as cms
+
+from Geometry.HcalCommonData.hcalSimulationParameters_cff   import *
+from Geometry.HcalCommonData.hcalSimulationConstants_cfi    import *
+from Geometry.HcalCommonData.hcalDBConstants_cff            import *
+from Geometry.HcalCommonData.caloSimulationParameters_cff   import *
+from Geometry.HcalCommonData.hcalDDDSimConstants_cfi        import *
+from Geometry.HcalCommonData.hcalDDDRecConstants_cfi        import *

